examples/tree_2d_dgsem/elixir_shallowwater_wall.jl

```diff
@@ -0,0 +1,83 @@
+using OrdinaryDiffEq
+using Trixi
+
+###############################################################################
+# Semidiscretization of the shallow water equations
+
+equations = ShallowWaterEquations2D(gravity_constant = 9.81, H0 = 3.25)
+
+# An initial condition with a perturbation in the waterheight to test boundary_condition_slip_wall
+function initial_condition_well_balancedness(x, t, equations::ShallowWaterEquations2D)
+    # Set the background values
+    H = equations.H0
+    v1 = 0.0
+    v2 = 0.0
+
+    x1, x2 = x
+    b = 0.0
```

Comment:
It would be good to set a non-constant bottom topography just to ensure that users see from this example elixir that the wall boundary condition works for generic bottom topographies.

Reply:
Yes, I have  added a bottom topography function and updated the test values in 7d03b4a65dfa508d5367916abd9a7497d1fa6407